src/main/frontend/src/app/appSlice.js

```diff
@@ -1,12 +1,24 @@
 import { createSlice } from '@reduxjs/toolkit';
+import { STATIC_COURSE_ARRAY } from './staticData.js';
 
-// Static data
-import { STATIC_COURSE_ARRAY } from './staticData';
+// Parse the LtiLaunchData JSON object from the root attribute.
+const ltiLaunchData = JSON.parse(document.getElementById('root').getAttribute('lti-launch-data'));
+
+// This defines the initial state of the store
+const initialState = {
+  searchInputText: '',
+  selectedCategoriesArray: [],
+  selectedCourse: null,
+  courseArray: STATIC_COURSE_ARRAY,
+  id_token: ltiLaunchData.id_token
+};
 
 // This defines the store slice
 export const appSlice = createSlice({
   // Name of the slice
   name: 'appSlice',
+  // Initial state
```

Comment:
The initial state is defined in index.js and it's already inserting the id_token